### PR TITLE
migrate code from googleapis/nodejs-asset

### PR DESCRIPTION
**Pull request by ahrarmonsur**
- initialize repo
- add a samples/package.json
- chore: do a little lint cleanup (#53)
- docs(samples): add quickstart code for ExportAssets (#50)
- refactor(samples): convert sample tests from ava to mocha (#56)
- fix: fix the lint rules (#58)
- sample: Add quickstart sample code for BatchGetAsstesHistory. (#59)
- Release v0.1.1 (#67)
- fix: fix sample tests (#79)
- Release v0.2.0 (#95)
- fix(deps): update dependency yargs to v13 (#105)
- chore(deps): update dependency mocha to v6
- Release v0.2.1 (#117)
- Release @google-cloud/asset v0.3.0 (#121)
- refactor: use execSync for tests (#127)
- feat: introduces 'expression text', deprecates version field (#143)
- docs: use repo-meta to generate README (#130)
- chore: release 1.0.0 (#146)
- chore: release 1.1.0 (#158)
- chore: release 1.1.1 (#161)
- chore: release 1.1.2 (#164)
- fix(deps): update dependency @google-cloud/storage to v3 (#168)
- chore: release 1.1.3 (#169)
- chore: release 1.1.4 (#176)
- fix(deps): update dependency yargs to v14
- feat: introduce v1p2beta1 client (#181)
- chore: release 1.2.0 (#182)
- docs: add real time feed api sample code (#184)
- chore: release 1.2.1 (#186)
- chore: release 1.3.0 (#191)
- chore: release 1.4.1 (#209)
- fix(deps): update dependency @google-cloud/storage to v4 (#207)
- chore: release 1.4.2 (#215)
- fix(deps): update dependency yargs to v15 (#222)
- chore: update license headers
- chore: release 1.5.0 (#225)
- feat(samples): update real time feed api sample code to use v1 API surface
- chore(deps): update dependency mocha to v7 (#241)
- chore: release 1.6.0
- chore: release 1.6.1 (#252)
- feat!: converts library to TypeScript adding v1p1beta1 surface (#250)
- chore: release 2.0.0 (#254)
- chore: release 2.0.1 (#262)
- chore: release 2.1.0 (#265)
- fix(deps): update dependency uuid to v7
- chore: release 2.2.0 (#271)
- fix: in rare cases test can take longer than 180,000 timeout (#307)
- chore: linting (#309)
- chore: release 3.0.0 (#280)
- fix(deps): update dependency uuid to v8 (#325)
- fix(deps): update dependency @google-cloud/storage to v5 (#330)
- chore: release 3.1.0 (#324)
- chore(deps): update dependency mocha to v8 (#341)
- docs(samples): add sample code for SearchAllResources and SearchAllIamPolicies (#343)
- chore: release 3.2.0 (#338)
- chore: release 3.3.0 (#347)
- chore: release 3.4.0 (#350)
- feat(samples): add sample code for ListAssets v1p5beta1 (#355)
- chore: release 3.5.0 (#356)
- chore: release 3.5.1 (#372)
- chore: release 3.6.0 (#382)
- chore: release 3.7.0 (#387)
- fix(deps): update dependency yargs to v16 (#391)
- chore: release 3.7.1 (#396)
- chore: release 3.8.0 (#399)
- chore: release 3.9.0 (#413)
- chore: release 3.9.1 (#425)
- chore: release 3.10.0 (#437)
- chore: release 3.11.0 (#446)
- feat(samples): add samples for analyzeIamPolicy and analyzeIamPolicyLongrunning (#433)
- docs: fix yaml parsing error in comments (#452)
- chore: release 3.12.0 (#450)
- chore: release 3.12.1 (#482)
- chore: release 3.12.2 (#493)
- chore: release 3.13.0 (#494)
- chore: release 3.14.0 (#506)
- samples: Update asset lib to v1 for ListAssets sample code (#505)
- chore: release 3.14.1 (#508)
- chore: release 3.15.0 (#516)
- chore: release 3.15.1 (#519)
- chore: release 3.15.2 (#521)
- chore: release 3.15.3 (#523)
- chore: release 3.15.4 (#535)
- chore: release 3.15.5 (#536)
- chore: release 3.16.0 (#539)
- chore: release 3.16.1 (#540)
- chore: release 3.17.0 (#543)
- chore: release 3.18.0 (#549)
- chore: release 3.19.0 (#552)
- samples: add relationship support samples (#560)
- docs(samples): add auto-generated samples for Node with api short name in region tag (#562)
- test: extend timeout for flaky tests (#567)
- chore: release 3.20.0 (#570)
- chore: release 3.21.0 (#576)
- chore: release 3.22.0 (#579)
- samples: Move sample code end tag to the correct line for GetFeed (#583)
- chore(main): release 3.23.0 (#603)
- chore(main): release 3.23.1 (#605)
- build!: update library to use Node 12 (#625)
- test: increase sample test timeout (#628)
- fix(deps): update dependency @google-cloud/bigquery to v6 (#630)
- fix(deps): update dependency @google-cloud/storage to v6 (#629)
- samples: fix compute test, update dependency to V3 (#634)
- chore(main): release 4.0.0 (#626)
- chore(main): release 4.1.0 (#636)
- chore(main): release 4.2.0 (#645)
- feat: add batchGetEffectiveIamPolicies sample code. (#654)
- fix(deps): update dependency uuid to v9 (#657)
- chore(main): release 4.3.0 (#652)
- chore(main): release 4.4.0 (#659)
- chore(main): release 4.5.0 (#671)

### CODE REVIEW

**Review by ace-n — CHANGES_REQUESTED** [review]
Can you add a workflow config to `.github/workflows` for this, so that the tests run?

> Be sure to add the config to `.github/workflows/workflows.json`, too. ✅

**Comment by ahrarmonsur** [comment]
The CI was failing due to lack of tests being carried over (the one mega-test in `samples.test.js` was marked as a quickstart test). However, it appears that one file was calling the tests for all the samples. 

On @grayside's advice to unblock this PR, I have:
- Created a tracking issue for language maintainers to refactor the tests to be sample-specific: https://github.com/GoogleCloudPlatform/nodejs-docs-samples/issues/2864
- Removed the workflow for `asset` samples

Context is in [this conversation](https://chat.google.com/room/AAAAQVhDuZk/k3xZ2L10Hrk)

File: asset/snippets/test/samples.test.js
Comment (ace-n) [inline asset/snippets/test/samples.test.js:1]: Did you mean to include this file?

**Review by ace-n — CHANGES_REQUESTED** [review]
Should these files be in `asset/snippets/`, or the `asset` directory itself?

File: asset/snippets/test/samples.test.js
Comment (ahrarmonsur) [reply asset/snippets/test/samples.test.js:1]: This singular test file includes not just the quickstart tests, but tests for all the other samples, as well. They are run sequentially.

I included it in the migration, so that when the language maintainers address the GH issue I created (https://github.com/GoogleCloudPlatform/nodejs-docs-samples/issues/2864), they would have the body of the individual tests already and could use their best judgment to modify them as needed to decompose into separate test files. At that point, I imagine the `samples.test.js` will be deleted.

I figure inclusion of this file here is acceptable tech debt for the migration. Since @sofisl already discarded most of the tests for her [migration to google-cloud-node](https://github.com/googleapis/google-cloud-node/blob/main/packages/google-cloud-asset/samples/test/sample.test.js), I didn't want to lose it in the migration/archival process.

Comment (ahrarmonsur) [reply asset/snippets/test/samples.test.js:1]: To address your other comment about migration destination:
i (and I believe most others) are following the example set in the migration guide, written by @grayside. You can see here that the samples end up `<product>/snippets`:
https://docs.google.com/document/d/1YhpsM5ylBFDunRsd-Vq7pHRNUz3uh9TUbOkJOZwCmOk/edit?resourcekey=0--XOS_x8NmzOS8_Qws-Qsug#bookmark=kix.z1viwa6f4215

**Review by ace-n — APPROVED** [review]
Omitting the `snippets` directory feels cleaner to me, **but** IMO it's not worth creating a new PR.

LGTM. ✅

**Pull request merged**
